streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/ClientState.java

```diff
@@ -53,16 +56,17 @@
     private final Map<TopicPartition, String> ownedPartitions = new TreeMap<>(TOPIC_PARTITION_COMPARATOR);
     private final Map<String, Set<TaskId>> consumerToPreviousStatefulTaskIds = new TreeMap<>();
 
-    // the following four maps are used only for logging purposes;
-    // TODO KAFKA-10283: we could consider merging them with other book-keeping maps at client-levels
-    //                   so that they would not be inconsistent
```

Comment:
I think maybe @guozhangwang and I haven't really explained the purpose of this cleanup ticket very well...it's Guozhang's ticket so I'll let him correct me if I've misunderstood the ticket as well, but my impression was that we wanted to consolidate the client-level sets -- eg `activeTasks` or `prevActiveTasks`, on lines 46-49 in the current code -- with the consumer-level maps, eg `consumerToPrevActiveTaskIds`. So the point is not so much to combine any of the maps on lines 59 - 62, but instead to combine each map with its corresponding set in the ones on lines 46 - 49. Does that make sense?

Reply:
Oh, I was misunderstanding
@ableegoldman 

Integrating two structures, for example

`class ClientStateTask {
   private final Set<TaskId> clientTaskIds
  private filnal Map<String, Set<TaskId>> consumerTaskIds
}`

Is this form correct?
thanks!

Reply:
@ableegoldman 
thanks for review
i updated code
add `class ClientStateTask`
Thanks!

Reply:
@ableegoldman yes you're right. On the high-level I want to consolidate sets that *always* get updated at the same time, since it is vulnerable to bugs when we forget to update some of them while modifying the others. But for maps that are logically orthogonal (i.e. they are not always updated at the same time) then we could still keep them separated.